influxdb2_client/src/api/label.rs

```diff
@@ -0,0 +1,305 @@
+            name: name.into(),
+            properties,
+        };
+        let response = self
+            .request(Method::POST, &create_label_url)
+            .body(serde_json::to_string(&body).context(Serializing)?)
+            .send()
+            .await
+            .context(ReqwestProcessing)?;
+        match response.status() {
+            StatusCode::CREATED => Ok(response
+                .json::<LabelResponse>()
+                .await
+                .context(ReqwestProcessing)?),
+            status => {
+                let text = response.text().await.context(ReqwestProcessing)?;
+                Http { status, text }.fail()?
+            }
+        }
+    }
+
+    /// Update a Label
+    pub async fn update_label(
+        &self,
+        name: Option<String>,
+        properties: Option<HashMap<String, String>>,
+        label_id: &str,
+    ) -> Result<LabelResponse, RequestError> {
+        let update_label_url = format!("{}/api/v2/labels/{}", &self.url, label_id);
+        let body = LabelUpdate { name, properties };
+        let response = self
+            .request(Method::PATCH, &update_label_url)
+            .body(serde_json::to_string(&body).context(Serializing)?)
+            .send()
+            .await
+            .context(ReqwestProcessing)?;
+        match response.status() {
+            StatusCode::OK => Ok(response
+                .json::<LabelResponse>()
+                .await
+                .context(ReqwestProcessing)?),
+            status => {
+                let text = response.text().await.context(ReqwestProcessing)?;
+                Http { status, text }.fail()?
+            }
+        }
+    }
+
+    /// Delete a Label
+    pub async fn delete_label(&self, label_id: &str) -> Result<(), RequestError> {
+        let delete_label_url = format!("{}/api/v2/labels/{}", &self.url, label_id);
+        let response = self
+            .request(Method::DELETE, &delete_label_url)
+            .send()
+            .await
+            .context(ReqwestProcessing)?;
+        match response.status() {
+            StatusCode::NO_CONTENT => Ok(()),
+            status => {
+                let text = response.text().await.context(ReqwestProcessing)?;
+                Http { status, text }.fail()?
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use mockito::mock;
+
+    type Error = Box<dyn std::error::Error>;
+    type Result<T = (), E = Error> = std::result::Result<T, E>;
+
+    const BASE_PATH: &str = "/api/v2/labels";
+
+    #[tokio::test]
+    async fn labels() -> Result {
+        let token = "some-token";
+
+        let mock_server = mock("GET", format!("{}?orgID=", BASE_PATH).as_str())
+            .match_header("Authorization", format!("Token {}", token).as_str())
+            .create();
+
+        let client = Client::new(&mockito::server_url(), token);
+
+        let _result = client.labels().await;
+
+        mock_server.assert();
+        Ok(())
+    }
```

Comment:
Ok, we decided in our meeting today that we're going to standardize on test functions not returning anything and using `expect` or `unwrap` in tests rather than `?`. So for example, this test should instead be:

```suggestion
    #[tokio::test]
    async fn labels() {
        let token = "some-token";

        let mock_server = mock("GET", format!("{}?orgID=", BASE_PATH).as_str())
            .match_header("Authorization", format!("Token {}", token).as_str())
            .create();

        let client = Client::new(&mockito::server_url(), token);

        let _result = client.labels().await;

        mock_server.assert();
    }
```

(there actually aren't any uses of `?` in these tests currently)

Please update this PR accordingly and then I'll take another look at it. I will take care of adding this convention to docs/style_guide.md and work on making the whole codebase consistent-- no need for you to go back and change past tests, just tests in this and future PRs :)

Reply:
okay will do that :)

Reply:
@carols10cents I was going to start working on a possible refactor, from data_point, write, errors splitting them into seperate files, possibly updating docs with examples -- and making sure there is a provision for having separate api versions (to accomadate v1.8 later on, if possible). Shall I drop that?

Reply:
@aakashhemadri I just [opened a PR](https://github.com/influxdata/influxdb_iox/pull/1120) with everything I was planning on doing regarding making the codebase consistent with regards to test function return types, so I'd wait til that gets merged in before starting any big reorganizing.

> I was going to start working on a possible refactor, from data_point, write, errors splitting them into seperate files, possibly updating docs with examples

That all sounds fine!

> making sure there is a provision for having separate api versions (to accomadate v1.8 later on, if possible).

We're actually not interested in having 1.8 support in this crate, given that it's called `influxdb*2*_client` :)

Reply:
Okayyy